sdk/internal/codegen/runner/generate.go

```diff
@@ -0,0 +1,245 @@
+		suffix:              "V2",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/entityresolution/v2",
+		packageNameOverride: "entityresolutionv2",
+	},
+	{
+		grpcClientInterface: "KeyAccessServerRegistryServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/policy/kasregistry",
+	},
+	{
+		grpcClientInterface: "KeyManagementServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/policy/keymanagement",
+	},
+	{
+		grpcClientInterface: "NamespaceServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/policy/namespaces",
+	},
+	{
+		grpcClientInterface: "RegisteredResourcesServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/policy/registeredresources",
+	},
+	{
+		grpcClientInterface: "ResourceMappingServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/policy/resourcemapping",
+	},
+	{
+		grpcClientInterface: "SubjectMappingServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/policy/subjectmapping",
+	},
+	{
+		grpcClientInterface: "UnsafeServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/policy/unsafe",
+	},
+	{
+		grpcClientInterface: "WellKnownServiceClient",
+		grpcPackagePath:     "github.com/opentdf/platform/protocol/go/wellknownconfiguration",
+	},
+}
+
+func Generate() error {
+	for _, client := range clientsToGenerateList {
+		slog.Info("Generating wrapper for", "interface", client.grpcClientInterface, "package", client.grpcPackagePath)
+		// Load the Go package using the import path
+		cfg := &packages.Config{
+			Mode: packages.NeedName |
+				packages.NeedTypes |
+				packages.NeedTypesInfo |
+				packages.NeedSyntax |
+				packages.NeedCompiledGoFiles,
+		}
+		pkgs, err := packages.Load(cfg, client.grpcPackagePath)
+		if err != nil {
+			return fmt.Errorf("failed to load package %s: %w", client.grpcPackagePath, err)
+		}
+		if packages.PrintErrors(pkgs) > 0 {
+			return fmt.Errorf("errors loading package %s", client.grpcPackagePath)
+		}
+		found := false
+		err = nil
+		// Loop through the package and its files
+		for _, p := range pkgs {
+			for _, file := range p.Syntax {
+				ast.Inspect(file, func(n ast.Node) bool {
+					if found {
+						return false // skip rest of traversal
+					}
+					ts, ok := n.(*ast.TypeSpec)
+					if !ok {
+						return true
+					}
+					iface, ok := ts.Type.(*ast.InterfaceType)
+					if !ok {
+						return true
+					}
+					if ts.Name.Name == client.grpcClientInterface {
+						packageName := path.Base(client.grpcPackagePath)
+						if client.packageNameOverride != "" {
+							packageName = client.packageNameOverride
+						}
+						code := generateWrapper(ts.Name.Name, iface, client.grpcPackagePath, packageName, client.suffix)
+						var currentDir string
+						currentDir, err = getCurrentFileDir()
+						outputPath := filepath.Join(currentDir, "..", "..", "..", "sdkconnect", packageName+".go")
```

Comment:
Maybe add an assertion about the output path to make sure it 'looks right', since this could be called from the 'wrong' folder

Reply:
thats the intention behind the currentfiledir() func, it gets the dir of this generate.go file so it will work no matter where its called from